File: src/test/java/eotree/data/TestEOCharData.java
package eotree.data;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

/**
 * EO char data tests.
 */
public class TestEOCharData {

    @Test
    public void testGenerateEOZeroIndent() {
        var c = new EOCharData('a');

        assertEquals(c.generateEO(0), "'a'");
    }

    @Test
    public void testGenerateEONonZeroIndent() {
        var c = new EOCharData('a');

        assertEquals(c.generateEO(1), "'a'");
    }
}

File: src/test/kotlin/eotree/data/TestEOCharData.kt
package eotree.data

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.polystat.j2eo.eotree.data.EOCharData

/**
 * EO char data tests.
 */
class TestEOCharData {
    @Test
    fun testGenerateEOZeroIndent() {
        val c = EOCharData('a')
        Assertions.assertEquals(c.generateEO(0), "'a'")
    }

    @Test
    fun testGenerateEONonZeroIndent() {
        val c = EOCharData('a')
        Assertions.assertEquals(c.generateEO(1), "'a'")
    }
}
